src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/EditorConfig/EditorConfigNamingStyleParser_SymbolSpec.cs

```diff
@@ -333,6 +333,9 @@ private static string ToEditorConfigString(this SymbolKindOrTypeKind symbol)
                 case TypeKind.Delegate:
                     return "delegate";
 
+                case TypeKind.TypeParameter:
```

Comment:
The change in this file and the following file was a bug that was found in the subsequent commit when tests were ported to code style layer. Attempting to serialize all possible combinations of naming styles to an editorconfig file (which is now done for the tests from code style layer) hits these cases.

Reply:
My strong preference in a case like this is to have this broken out into its own small PR that can then go through independently.  This PR canthen branchoff of those.

Reply:
That was my original intent, but unfortunately the naming tests need to be in CodeStyle layer to fail due to this change, and that is done in the last commit of this PR.

Reply:
@CyrusNajmabadi I was actually able to find a simpler unit test that uncovers these failures. So, I have extracted this bug fix into https://github.com/dotnet/roslyn/pull/42245. Thanks!